### ISSUE TITLE
docs: `endpoint_name` in  `mongodbatlas_privatelink_endpoint_service` should link to the name of the forwarding_rule, not address

### ISSUE THREAD

**Issue by TJM** [mongodb/terraform-provider-mongodbatlas#753]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
# Terraform v1.1.5 (not relevant)
```

### Terraform Configuration File 

```hcl
# N/A
```

### Steps to Reproduce

- Docs Example for GCP:  https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint_service

If you name the IP address entries differently than the forwarding rule entries, it fails,

### Expected Behavior

The example should use the correct link for endpoint_name

```hcl
resource "mongodbatlas_privatelink_endpoint_service" "test" {
  project_id          = mongodbatlas_privatelink_endpoint.test.project_id
  private_link_id     = mongodbatlas_privatelink_endpoint.test.private_link_id
  provider_name       = "GCP"
  endpoint_service_id = google_compute_network.default.name
  gcp_project_id      = var.gcp_project

  dynamic "endpoints" {
    for_each = mongodbatlas_privatelink_endpoint.test.service_attachment_names

    content {
      ip_address    = google_compute_address.default[endpoints.key].address
      endpoint_name = google_compute_forwarding_rule.default[endpoints.key].name
    }
  }

  depends_on = [google_compute_forwarding_rule.default]
}
```

### Actual Behavior

It uses the name of the IP address instead of the forwarding rule.

```hcl
  dynamic "endpoints" {
    for_each = mongodbatlas_privatelink_endpoint.test.service_attachment_names

    content {
      ip_address    = google_compute_address.default[endpoints.key].address
      endpoint_name = google_compute_address.default[endpoints.key].name
    }
  }
```

### Debug Output 

N/A - doc bug

### Crash Output

N/A

### Additional Context

The example just needs fixed.

**Comment by themantissa** [comment]
Thank you @TJM - we'll check this and get it corrected.

**Comment by themantissa** [comment]
Internal ticket INTMDB-331

**Comment by martinstibbe** [comment]
@TJM This is corrected in next release 1.4.4

**Comment by themantissa** [comment]
@TJM the release is out (cc @martinstibbe )
